### PR TITLE
fix: remove old web platform distributions on VERSION update

### PR DESCRIPTION
**Pull request by itayd**
when running `make ak`, properly discard old downloaded versions. currently it doesn't because bash is HARD, which causes the server to possibly use the wrong version.

also:
- alert and disable webplatform when used version does not match VERSION, do not use older versions.
- specifying the sha is no longer mandatory.
- upgrade to latest web-platform.

### CODE REVIEW

File: web/webplatform/fetch.sh
Comment (tebeka) [inline web/webplatform/fetch.sh:1]: IMO this file is complex enough to merit a rewrite in Go.

File: web/webplatform/fetch.sh [hunk 1]
```diff
@@ -4,13 +4,18 @@ set -euo pipefail
...
 
 checksum() {
+    if [[ -z "${EXPECTED_SHA}" ]]; then
+        echo "no checksum provided"
+        return
+    fi
+
     echo "verifying checksum..."
 
     sha=$(shasum -a 256 "$1" | cut -d ' ' -f 1)
```
Comment (tebeka) [inline web/webplatform/fetch.sh]: This works only because it's the first field. Use awk here as well?

**Review by daabr — APPROVED** [review]

File: web/webplatform/webplatform.go [hunk 2]
```diff
@@ -27,17 +30,35 @@ func DistFilename() (string, error) {
 		return "", err
 	}
 
-	_, de := kittehs.FindFirst(des, func(de fs.DirEntry) bool { return distRegex.MatchString(de.Name()) })
-	if de == nil {
+	des = kittehs.Filter(des, func(de fs.DirEntry) bool { return distRegex.MatchString(de.Name()) })
```
Comment (tebeka) [inline web/webplatform/webplatform.go]: Why do we need to glob? You can know the exact file name from `VERSION`

File: web/webplatform/webplatform_test.go [hunk 3]
```diff
@@ -6,12 +6,13 @@ import (
...
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 
 	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
 )
 
 func TestLoad(t *testing.T) {
-	fs, v, err := LoadFS()
+	fs, v, err := LoadFS(zap.NewNop())
```
Comment (tebeka) [inline web/webplatform/webplatform_test.go]: This will quell errors, use `zaptest`?

File: web/webplatform/webplatform_test.go [hunk 4]
```diff
@@ -27,3 +28,20 @@ func TestLoad(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, f)
 }
+
+func TestVersion(t *testing.T) {
+	l, _ := zap.NewDevelopment() // if test fails, will show detailed version error.
```
Comment (tebeka) [inline web/webplatform/webplatform_test.go]: Check for error, otherwise test failures will be hard to understand.

**Review by tebeka — APPROVED** [review]
Some minor comment but can merge.

File: web/webplatform/fetch.sh
Comment (itayd) [reply web/webplatform/fetch.sh:1]: if we take it further somehow, will consider.

File: web/webplatform/webplatform.go (hunk 2 above)
Comment (itayd) [reply web/webplatform/webplatform.go]: oh wow, right. will try.

File: web/webplatform/fetch.sh (hunk 1 above)
Comment (itayd) [reply web/webplatform/fetch.sh]: done

File: web/webplatform/webplatform_test.go (hunk 3 above)
Comment (itayd) [reply web/webplatform/webplatform_test.go]: done

File: web/webplatform/webplatform_test.go (hunk 4 above)
Comment (itayd) [reply web/webplatform/webplatform_test.go]: removed test

**Comment by itayd** [comment]
greatly simplified the code since now only using version in VERSION. Thanks @tebeka for the tip. going to push this in, lmk if any issues - I'll fix.

**Pull request merged**